webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/nodes/NewNodeManager.scala

```diff
@@ -185,7 +182,7 @@ class NewNodeManagerHooksImpl(
     .runNow
 
   override def afterNodeAcceptedAsync(nodeId: NodeId)(implicit qc: QueryContext): IOResult[Unit] = {
-    codeHooks.get.flatMap(hooks => ZIO.foreach(hooks)(h => IOResult.attempt(h.run(nodeId)))).unit
+    codeHooks.get.flatMap(hooks => ZIO.foreach(hooks)(h => IOResult.attempt(h.run(nodeId).forkDaemon.runNow))).unit
```

Comment:
you can can't `.runNow` in a foreach. 
But I think that the root cause: `.run` is now an IOResult, so it's just a no-op in the `attempt`. `IOResult.attempt` need to be removed.